### PR TITLE
python3Packages.scikitlearn: rename to scikit-learn

### PR DESCRIPTION
**Pull request by DavHau**
###### Motivation for this change
The  name on pypi for this package is `scikit-learn` but it is named `scikitlearn` in nixpkgs

###### Things done
Correct the name while staying downwards conpatible.

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 123017` at 0351e84f run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.scikit-learn</li>
    <li>python39Packages.scikit-learn</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/scikit-learn/default.nix:69:0:
```
   |
69 |     description = "A set of python modules for machine learning and data mining";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/scikit-learn/default.nix:78:5:
```
   |
78 |     maintainers = with maintainers; [ ];
   |     ^
```
</li>
  </ul>
</details>

**Comment by rmcgibbo** [comment]
Should we also rename it in all downstream packages within nixpkgs at the same time?

File: pkgs/top-level/python2-packages.nix [hunk 1]
```diff
@@ -524,10 +524,12 @@ with self; with super; {
 
   scandir = callPackage ../development/python-modules/scandir { };
 
-  scikitlearn = callPackage ../development/python-modules/scikitlearn/0.20.nix {
+  scikit-learn = callPackage ../development/python-modules/scikit-learn/0.20.nix {
     inherit (pkgs) gfortran glibcLocales;
   };
 
+  scikitlearn = python.pkgs.scikit-learn;
```
Comment (dotlambda) [inline pkgs/top-level/python2-packages.nix:531]:
```suggestion
  scikitlearn = self.scikit-learn;
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by DavHau** [comment]
Committed suggestion and also enabled parallel building and testing

File: pkgs/top-level/python2-packages.nix [hunk 2]
```diff
@@ -524,10 +524,12 @@ with self; with super; {
 
   scandir = callPackage ../development/python-modules/scandir { };
 
-  scikitlearn = callPackage ../development/python-modules/scikitlearn/0.20.nix {
+  scikit-learn = callPackage ../development/python-modules/scikit-learn/0.20.nix {
     inherit (pkgs) gfortran glibcLocales;
   };
 
+  scikitlearn = self.scikit-learn;
```
Comment (dotlambda) [inline pkgs/top-level/python2-packages.nix:531]: I think this shouldn't be necessary.

**Review by dotlambda — CHANGES_REQUESTED** [review]
Can you squash your commits into two commits: one for the renaming and one for enabling the tests?

**Comment by DavHau** [comment]
squashed commits and removed `scikitlearn` alias.

File: pkgs/development/python-modules/scikit-learn/default.nix [hunk 3]
```diff
@@ -54,17 +55,24 @@ buildPythonPackage rec {
...
+  preBuild = ''
+    export SKLEARN_BUILD_PARALLEL=$(( $(nproc) + 1 ))
+  '';
+
   doCheck = !stdenv.isAarch64;
   # Skip test_feature_importance_regression - does web fetch
   checkPhase = ''
     cd $TMPDIR
-    HOME=$TMPDIR OMP_NUM_THREADS=1 pytest -k "not test_feature_importance_regression" --pyargs sklearn
+    HOME=$TMPDIR OMP_NUM_THREADS=1 pytest -n $(( $(nproc) + 1 )) -k "not test_feature_importance_regression" --pyargs sklearn
```
Comment (dotlambda) [inline pkgs/development/python-modules/scikit-learn/default.nix:71]:
```suggestion
    HOME=$TMPDIR OMP_NUM_THREADS=1 pytest -n $NIX_BUILD_CORES -k "not test_feature_importance_regression" --pyargs sklearn
```

File: pkgs/development/python-modules/scikit-learn/default.nix [hunk 4]
```diff
@@ -54,17 +55,24 @@ buildPythonPackage rec {
     joblib
     threadpoolctl
   ];
-  checkInputs = [ pytest ];
+
+  checkInputs = [ pytest pytest-xdist ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/scikit-learn/default.nix:59]:
```suggestion
  checkInputs = [ pytestCheckHook pytest-xdist ];
```
You'll have to use `disabledTests` and `pytestFlagsArray`.

**Review by dotlambda — COMMENTED** [review]
I fixed the commit messages and removed a commit involving prosody.

File: pkgs/top-level/python2-packages.nix (hunk 2 above)
Comment (rmcgibbo) [reply pkgs/top-level/python2-packages.nix]: Why not? Won’t changing the name without adding an alias potentially break out-of-tree user nix code, flakes, etc?

Comment (dotlambda) [reply pkgs/top-level/python2-packages.nix]: @DavHau I didn't notice you removed it from `python-packages.nix` as well. I just meant the line in `python2-packages.nix` shouldn't be necessary.

@rmcgibbo Any alias in `python-packages.nix` will be available for both Python 2 and Python 3.

File: pkgs/development/python-modules/scikit-learn/default.nix (hunk 3 above)
Comment (DavHau) [reply pkgs/development/python-modules/scikit-learn/default.nix]: done

File: pkgs/development/python-modules/scikit-learn/default.nix (hunk 4 above)
Comment (DavHau) [reply pkgs/development/python-modules/scikit-learn/default.nix]: done

File: pkgs/development/python-modules/scikit-learn/default.nix [hunk 5]
```diff
@@ -54,17 +55,30 @@ buildPythonPackage rec {
...
 
+  preBuild = ''
+    export SKLEARN_BUILD_PARALLEL=$NIX_BUILD_CORES
+  '';
+
   doCheck = !stdenv.isAarch64;
+
   # Skip test_feature_importance_regression - does web fetch
-  checkPhase = ''
+  disabledTests = "test_feature_importance_regression";
```
Comment (dotlambda) [inline pkgs/development/python-modules/scikit-learn/default.nix:70]:
```suggestion
  disabledTests = [ "test_feature_importance_regression" ];
```

Comment (DavHau) [reply pkgs/development/python-modules/scikit-learn/default.nix]: done

**Comment by dotlambda** [comment]
@ofborg build python38.pkgs.scikit-learn python39.pkgs.scikit-learn

**Comment by dotlambda** [comment]
On Darwin, the tests also failed before this change, so that can be fixed another time.

**Pull request merged**